tsconfig.json

```diff
@@ -1,10 +1,36 @@
 {
+  // see https://www.typescriptlang.org/tsconfig to better understand tsconfigs
+  "include": ["src", "types"],
   "compilerOptions": {
+    "module": "esnext",
+    "lib": ["dom", "esnext"],
+    "importHelpers": true,
+    // output .d.ts declaration files for consumers
     "declaration": true,
-    "lib": ["DOM", "ESNext"],
-    "module": "ESNext",
+    // output .js.map sourcemap files for consumers
+    "sourceMap": true,
+    // match output dir to input dir. e.g. dist/index instead of dist/src/index
+    "rootDir": "./src",
+    // stricter type-checking for stronger correctness. Recommended by TS
     "strict": true,
-    "target": "ES6"
-  },
-  "files": ["lib/pin-field.ts"]
+    // linter checks for common issues
+    "noImplicitReturns": true,
+    "noFallthroughCasesInSwitch": true,
+    // noUnused* overlap with @typescript-eslint/no-unused-vars, can disable if duplicative
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "target": "esnext",
```

Comment:
Sure about this one? Not all the browsers are understanding `ESNext` right now, I would stick with `ES6` (see https://www.typescriptlang.org/tsconfig#target):

> Modern browsers support all ES6 features, so ES6 is a good choice.